src/com/algorithm/exactcover/nodes/DancingNode.java

```diff
@@ -0,0 +1,105 @@
+package com.algorithm.exactcover.nodes;
+
+/**
+ * This class represent the dancing node
+ *
+ * @author Jose Cabrera
+ */
+public class DancingNode{
+    protected DancingNode left, right, up, down;
+    protected ColumnNode columnNode;
+
+    public DancingNode(){
+        left = right = up = down = this;
+    }
+
+    public DancingNode(ColumnNode columnNode){
+        this();
+        setColumnNode(columnNode);
+    }
+
+    public ColumnNode getColumnNode() {
+        return columnNode;
+    }
+
+    public void setColumnNode(ColumnNode columnNode) {
+        this.columnNode = columnNode;
+    }
+
+    public DancingNode getLeft() {
+        return left;
+    }
+
+    public DancingNode getRight() {
+        return right;
+    }
+
+    public DancingNode getUp() {
+        return up;
+    }
+
+    public DancingNode getDown() {
+        return down;
+    }
+
+    /**
+     * hooks node dancingNode below current node
+     * @param dancingNode
+     * @return
+     */
+    public DancingNode hookDown(DancingNode dancingNode){
+        assert (getColumnNode() == dancingNode.getColumnNode());
```

Comment:
if the assertion is not correct it will skip the other lines of code is this correct?